components/scream/src/physics/spa/atmosphere_prescribed_aerosol.cpp

```diff
@@ -155,14 +175,21 @@ void SPA::initialize_impl (const RunType /* run_type */)
   SPAHorizInterp.m_comm = m_comm;
 
   // Initialize the size of the SPAData structures:
-  SPAData_start = SPAFunc::SPAData(m_dofs_gids.size(), source_data_nlevs, m_nswbands, m_nlwbands);
-  SPAData_end   = SPAFunc::SPAData(m_dofs_gids.size(), source_data_nlevs, m_nswbands, m_nlwbands);
+  SPAData_start = SPAFunc::SPAInput(m_dofs_gids.size(), source_data_nlevs, m_nswbands, m_nlwbands);
+  SPAData_end   = SPAFunc::SPAInput(m_dofs_gids.size(), source_data_nlevs, m_nswbands, m_nlwbands);
 
   // Update the local time state information and load the first set of SPA data for interpolation:
   auto ts = timestamp();
   SPATimeState.inited = false;
   SPATimeState.current_month = ts.get_month();
   SPAFunc::update_spa_timestate(m_spa_data_file,m_nswbands,m_nlwbands,ts,SPAHorizInterp,SPATimeState,SPAData_start,SPAData_end);
+
+  // NOTE: we *assume* hybrid v coordinates don't change with time.
+  //       IF this ever ceases to be the case, you need to remove these
+  //       lines, and have spa_main interpolate those during the call
+  //       to performe_time_interpolation.
```

Comment:
I suspect that if we ever allow the hybrid coordinates to change with time then we would just store them in SPAData_start and SPAData_end and when we construct pmid_src we would have to first construct pmid_start and pmid_end with the respective hya/b and then do time_interpolation between those two.

Not that I see that happening anytime soon.